core/controllers/retrieve.py

```diff
@@ -0,0 +1,34 @@
+from io import BytesIO
+
+import flask
+from flask import abort, request
+from sqlalchemy import select
+from sqlalchemy.exc import NoResultFound
+
+from core.const import EXCEL_MIMETYPE
+from core.db import db
+from core.db.entities import Submission
+
+
+def retrieve():
+    """Handle the download request and return the originally submitted spreadsheet.
+
+    Select file by:
+    - submission_id
+
+    :return: Flask response object containing the requested spreadsheet.
+    """
+    submission_id = request.args.get("submission_id")
```

Comment:
Using Connexion, we can put `submission_id` as a function param and it will grab it from the request rather than having to fetch it from the `Flask` request object.

See https://connexion.readthedocs.io/en/stable/request.html